core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -39,15 +38,12 @@
 	@Autowired
 	private AnalysisEngineCacheService cacheService;
 	@Autowired
-	private UserService userService;
-	@Autowired
 	private UserAnonymizedService userAnonymizedService;
 	@Autowired
 	private MessageService messageService;
 
-	public void analyze(UUID userID, AppActivityDTO[] appActivities)
+	public void analyze(UUID userAnonymizedID, AppActivityDTO[] appActivities)
 	{
-		UUID userAnonymizedID = userService.getPrivateUser(userID).getPrivateData().getUserAnonymizedID();
 		UserAnonymizedDTO userAnonymized = userAnonymizedService.getUserAnonymized(userAnonymizedID);
 		for (AppActivityDTO appActivity : appActivities)
```

Comment:
Yes, I agree. If the posted app activity is in a day before the current cached day, the day should be fetched from the database.